packages/tables/src/Filters/Concerns/HasRelationship.php

```diff
@@ -31,6 +31,8 @@ public function getRelationshipKey(): string
             $keyColumn = $relationship->getQualifiedRelatedKeyName();
         } elseif ($relationship instanceof HasOneThrough) {
             $keyColumn = $relationship->getQualifiedForeignKeyName();
+        } elseif ($relationship instanceof \Znck\Eloquent\Relations\BelongsToThrough) {
+            $keyColumn = $relationship->getRelated()->getKeyName();
```

Comment:
```suggestion
            $keyColumn = $relationship->getRelated()->getQualifiedKeyName();
```

Please check the integration still works after making this change